### PR TITLE
[next] Refresh all config

### PR DESCRIPTION
**Pull request by JillieBeanSim**
## Proposed changes

Silently reload Zowe Explorer to reset which treeProviders are used after types of profile handling are changed, old style => config, config -> old sytle, etc.  User is presented with a button in the Popup Message after config file creation to `Refresh Zowe Explorer` other options for this action are the Command Palette `Zowe Explorer: Refresh Zowe Explorer` and quick keys `alt+z`

refresh-zowe-explorer

## Release Notes

Milestone:

Changelog:

## Types of changes

What types of changes does your code introduce to Zowe Explorer?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Updates to Documentation or Tests (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This checklist will be used as reference for both the contributor and the reviewer_

- [x] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/vscode-extension-for-zowe/wiki/Best-Practices:-Contributor-Guidance) wiki
- [x] PR title follows [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0-beta.2/)
- [x] PR Description is included
- [x] gif or screenshot is included if visual changes are made
- [x] `yarn workspace vscode-extension-for-zowe vscode:prepublish` has been executed
- [ ] All checks have passed (DCO, Jenkins and Code Coverage)
- [ ] I have added unit test and it is passing
- [ ] I have added integration test and it is passing
- [ ] There is coverage for the code that I have added
- [ ] I have tested it manually and there are no regressions found
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any PR dependencies have been merged and published (if appropriate)

## Reminder

After a PR is merged into the `master` branch, create a PR from `master` to the `next` branch resolving any conflicts.

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by crawr** [comment]
Refresh all for new config file:
create new config

Refresh all for added profile:
add new profile

**Comment by JillieBeanSim** [comment]
Since it isn't refreshing automatically after creation and it will be good to notify the user of the location of the created config and letting the user to have the chance to update it before refreshing the views to avoid errors I think we should add the pop up info message back.

**Comment by JillieBeanSim** [comment]
@crawr I am still seeing the issues that I mentioned in the description, it is due to the initial registration/activation being attached to the old treeProviders.  I do have a solution that will silently reload Zowe Explorer and initialize the new treeProviders.  This will not touch the View's refresh buttons and will require another action to `Refresh Zowe Explorer` by either the pop up message button, command palete, or a quick key option.  I will put together a PR into this branch for us to test and see if this would be a better option.

**Comment by crawr** [comment]
I ran the Theia tests locally and all 22 are passing for me. Not sure why they are failing via GitHub actions. The CI builds are failing due to timeout, will try to rerun after some time.

**Comment by JillieBeanSim** [comment]
> I ran the Theia tests locally and all 22 are passing for me. Not sure why they are failing via GitHub actions. The CI builds are failing due to timeout, will try to rerun after some time.

They are failing on master too, ever since we started grabbing Theia 1.22.1.  I tested it manually and didn't see any issues before the release last week.  I did create an issue for investigation but the IDs and Xpaths haven't changed or anything so not sure why they are failing in the CI. https://github.com/zowe/vscode-extension-for-zowe/issues/1665

**Comment by JillieBeanSim** [comment]
@crawr here is the new approach https://github.com/zowe/vscode-extension-for-zowe/pull/1673

**Review by rudyflores — APPROVED** [review]
LGTM! I tested the PR by having existing V1 profiles and generating a config file for v2 profiles, hit the refresh button and saw the default profile appear! I also tested adding a new profile in the config and hitting reload, and as well deleting the config file and schema to revert to v1 profiles and they both work as intended :)

**Comment by JillieBeanSim** [comment]
I am working on updating the `Early Access -Using Global Profile Configuration.md` to reflect these changes as well as others in this release and will include them in the release PR since the compatible Zowe CLI chart will require the Zowe Explorer release version number in it.

File: packages/zowe-explorer/src/shared/refresh.ts [hunk 1]
```diff
@@ -26,6 +29,8 @@ import * as globals from "../globals";
  * @param {IZoweTree} treeProvider
  */
 export async function refreshAll(treeProvider: IZoweTree<IZoweTreeNode>) {
+    await readConfigFromDisk();
+    await Profiles.getInstance().refresh(ZoweExplorerApiRegister.getInstance());
```
Comment (zFernand0) [inline packages/zowe-explorer/src/shared/refresh.ts:33]: This makes the extension work seamlessly with the CLI
Whether that's doing
- ZE: Log in to Authentication Service  -> CLI commands
- CLI auth login OR CLI config auto-init -> ZE browsing (after a soft refresh using the button)

File: packages/zowe-explorer/src/extension.ts [hunk 2]
```diff
@@ -161,66 +175,84 @@ export async function activate(context: vscode.ExtensionContext): Promise<ZoweEx
         initJobsProvider(context, jobsProvider);
     }
     if (datasetProvider || ussFileProvider) {
-        vscode.commands.registerCommand("zowe.openRecentMember", () =>
-            sharedActions.openRecentMemberPrompt(datasetProvider, ussFileProvider)
+        context.subscriptions.push(
```
Comment (zFernand0) [inline packages/zowe-explorer/src/extension.ts:178]: From what I'm gathering, all registered commands are now part of the context subscriptions.
That way the commands are properly de-registered by VSCode when the extension is removed/disabled
Is that the intention here as well ?

Source: https://stackoverflow.com/a/55579618/16555983

**Review by zFernand0 — APPROVED** [review]
LGTM! 
Thank you @JillieBeanSim and @crawr 😋

File: packages/zowe-explorer/src/extension.ts [hunk 3]
```diff
@@ -122,6 +122,20 @@ export async function activate(context: vscode.ExtensionContext): Promise<ZoweEx
...
+            for (const sub of context.subscriptions) {
+                try {
+                    await sub.dispose();
+                } catch (e) {
+                    globals.LOG.error(e);
+                }
+            }
+            await activate(context);
+        })
+    );
```
Comment (JillieBeanSim) [inline packages/zowe-explorer/src/extension.ts:138]: @zFernand0 the subscriptions to all the other commands allow the loop here to dispose of them before reactivating.  If they aren't disposed errors will pop up stating that the commands are already registered and in our case they are registered with the previous treeProvider.

Comment (zFernand0) [reply packages/zowe-explorer/src/extension.ts:138]: thanks! 😋 
that helps! : )

**Pull request merged**